core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -417,7 +418,7 @@ class MapUnit {
         tile.improvementInProgress = null
     }
 
-    private fun tryProvideProductionToClosestCity() {
+    private fun tryProvideProductionToClosestCity(cause: String) {
```

Comment:
For better readability call this removedTerrainFeature or something, "cause" is very vague